sdv/constraints/tabular.py

```diff
@@ -642,13 +642,36 @@ def _get_diff_column_name(self, table_data):
 
         return token.join(components)
 
+    def _get_is_datetime(self, table_data):
+        low = self._get_low_value(table_data)
+        high = self._get_high_value(table_data)
+        column = table_data[self.constraint_column]
+
+        is_low_datetime = (pd.api.types.is_datetime64_any_dtype(low)
+                           or isinstance(low, pd.Timestamp)
+                           or isinstance(low, datetime))
+        is_high_datetime = (pd.api.types.is_datetime64_any_dtype(high)
+                            or isinstance(high, pd.Timestamp)
+                            or isinstance(high, datetime))
+        is_column_datetime = (pd.api.types.is_datetime64_any_dtype(high)
```

Comment:
@csala `is_datetime64_any_dtype` should work for any datetime dtype. I preferred this over the `dtype.kind` check, because doing `low.dtype.kind` will error out if `low` is a scalar.